allennlp/training/metrics/rouge.py

```diff
@@ -148,6 +156,18 @@ def _get_rouge_n_stats(
             total_precision += precision
             total_f1 += f1
 
+        if is_distributed():
+            device = predicted_tokens.device
+            _total_recall = torch.tensor(total_recall).to(device)
+            _total_precision = torch.tensor(total_precision).to(device)
+            _total_f1 = torch.tensor(total_f1).to(device)
+            dist.all_reduce(_total_recall, op=dist.ReduceOp.SUM)
+            dist.all_reduce(_total_precision, op=dist.ReduceOp.SUM)
+            dist.all_reduce(_total_f1, op=dist.ReduceOp.SUM)
+            total_recall = _total_recall.item()
+            total_precision = _total_precision.item()
+            total_f1 = _total_f1.item()
```

Comment:
This ends up returning the sum of the F1 scores? Maybe I'm missing context, but why is it OK to sum up the F1 scores?

Reply:
The calculation for getting the precision, recall, f1-scores does the following in the for loop:

```
# Accumulate stats
total_recall += recall
total_precision += precision
total_f1 += f1
```

In the distributed setting, we sum the sums.

Reply:
I'll assume Tobi had his reasons for writing it this way then.